packages/file-storage/src/file-storage-s3.class.ts

```diff
@@ -113,80 +126,93 @@ export class FileStorageS3 implements FileStorage {
     const { filePath, options = {}, request } = args;
     const { s3, bucket: Bucket } = this.config;
     const Key = await this.transformFilePath(filePath, MethodTypes.READ, request, options);
-    await s3.headObject({ Key, Bucket, ...options }).promise();
+    await s3.headObject({ Key, Bucket, ...options });
     return true;
   }
 
   async uploadFile(args: FileStorageS3UploadFile): Promise<void> {
     const { filePath, content, options = {}, request } = args;
     const { s3, bucket: Bucket } = this.config;
     const Key = await this.transformFilePath(filePath, MethodTypes.WRITE, request, options);
-    await s3.upload({ Bucket, Key, Body: content, ...options }).promise();
+    await new Upload({
+      client: s3,
+      params: { Bucket, Key, Body: content, ...options },
+    }).done();
   }
 
   async uploadStream(args: FileStorageS3UploadStream): Promise<Writable> {
     const { filePath, options = {}, request } = args;
     const Key = await this.transformFilePath(filePath, MethodTypes.WRITE, request, options);
     const { s3, bucket: Bucket } = this.config;
     const writeStream = new PassThrough();
-    s3.upload({
-      Body: writeStream,
-      Key,
-      Bucket,
-      ...options,
-    }).send((err) => writeStream.destroy(err));
-    //? or s3.upload({ Bucket, Key, Body: writeStream, ...options }, (err) => {
-    //   if (err) console.error(err);
-    // });
+    try {
+      new Upload({
+        client: s3,
+        params: {
+          Body: writeStream,
+          Key,
+          Bucket,
+          ...options,
+        },
+      }).done();
+    } catch (err) {
+      writeStream.destroy(err);
+    }
```

Comment:
i think this catch is useless because `new Upload` returns a promise so if it rejects, the error would be unhandled in that setup.

Reply:
this would probably be the way to go:
```ts
    new Upload({
      client: s3,
      params: {
        Body: writeStream,
        Key,
        Bucket,
        ...options,
      },
    })
      .done()
      .catch((err) => {
        writeStream.destroy(err);
      });
```